### PR TITLE
[Feature launch] Re-add increment operator

### PR DESCRIPTION
**Pull request by inlined**
For eventually merging fixes after adding new tests.

### CODE REVIEW

File: Firebase/Database/Core/FServerValues.m [hunk 1]
```diff
@@ -69,22 +145,28 @@ + (id)resolveDeferredValueTree:(FSparseSnapshotTree *)tree
                              rememberData:
                                  [FServerValues
                                      resolveDeferredValueSnapshot:node
-                                                 withServerValues:serverValues]
+                                                     withExisting:
+                                                         [existing
+                                                             getChild:path]
```
Comment (inlined) [inline Firebase/Database/Core/FServerValues.m]: This `[existing getChild:path]` is the line that was missing from the JS SDK. We could add the rejected performance optimizations with a lazy-evaluation of data, but then we should check it into JS too.

**Comment by inlined** [comment]
Feature branch. I'll rename

**Comment by inlined** [comment]
Should be ready for final review @schmidt-sebastian

File: Example/Database/Tests/Integration/FData.m [hunk 2]
```diff
@@ -2873,6 +2874,137 @@ - (void)testServerValuesTransactionLocalEvents {
...
+  [ref setValue:@{@"hello" : @"world"}];
+  [ref setValue:[FIRServerValue increment:@1]];
+  [expected addObject:@{@"hello" : @"world"}];
+  [expected addObject:@1];
+
+  [self waitUntil:^BOOL {
+    return found.count == expected.count;
+  }];
+  XCTAssertEqualObjects(expected, found);
+  [ref.repo resume];
```
Comment (schmidt-sebastian) [inline Example/Database/Tests/Integration/FData.m]: Should we add online versions of all these tests?

File: Example/Database/Tests/Integration/FData.m [hunk 3]
```diff
@@ -2873,6 +2874,137 @@ - (void)testServerValuesTransactionLocalEvents {
...
+  [expected addObject:@2.5];
+
+  [self waitUntil:^BOOL {
+    return found.count == expected.count;
+  }];
+  XCTAssertEqualObjects(expected, found);
+  [ref.repo resume];
+}
+
+- (void)testServerIncrementOverflowAndTypeCoercion {
```
Comment (inlined) [inline Example/Database/Tests/Integration/FData.m]: This one doesn't make sense to have online behavior; that's totally redundant with server tests (we're trying to test local coercion)

File: Example/Database/Tests/Integration/FData.m (hunk 2 above)
Comment (inlined) [reply Example/Database/Tests/Integration/FData.m]: I'll do this and the next test

Comment (inlined) [reply Example/Database/Tests/Integration/FData.m]: May break Travis; I'm not able to run tests locally.

File: Firebase/Database/Core/FServerValues.m [hunk 4]
```diff
@@ -117,49 +183,54 @@ + (id)resolveComplexServerOp:(NSDictionary *)op
...
                              serverValues:(NSDictionary *)serverValues {
+    id<ValueProvider> jitExisting =
+        [[ExistingValueProvider alloc] initWithSnapshot:existing];
+    return [FServerValues resolveDeferredValueSnapshot:node
+                                       withJITExisting:jitExisting
+                                          serverValues:serverValues];
+}
+
++ (id<FNode>)resolveDeferredValueSnapshot:(id<FNode>)node
+                          withJITExisting:(id<ValueProvider>)existing
```
Comment (schmidt-sebastian) [inline Firebase/Database/Core/FServerValues.m:229]: Nit: Possibly rename to `withValueProvider`.

Comment (inlined) [reply Firebase/Database/Core/FServerValues.m:229]: 🤦‍♂️You're right, of course. Guess I'll add another commit and wait another hour to check in.

**Pull request merged**